monster.py

```diff
@@ -2,31 +2,22 @@
 
 from moving_thing import MovingThing
 from abc import ABCMeta, abstractmethod
+from collections import namedtuple
 
+Point = namedtuple('Point', ['y', 'x'])
 
 class Monster(MovingThing, metaclass=ABCMeta):
     def follow_player(self):
-        nx, ny = self.x, self.y
-        if self.y == self.room.player.y and self.room.player.x == self.x:
-            return
-
-        if self.room.player.x > self.x:
-            nx = self.x + 1
-        elif self.room.player.x < self.x:
-            nx = self.x - 1
-
-        if self.room.player.y > self.y:
-            ny = self.y + 1
-        elif self.room.player.y < self.y:
-            ny = self.y - 1
-
-        if self.x != nx and self.y != ny:
-            if random.randrange(2):
-                nx = self.x
-            else:
-                ny = self.y
-
-        self.start_moving(nx, ny)
+        neighbours = [n for n in {Point(self.y - 1, self.x),
```

Comment:
v tomto pripade je to mnozina, spravil som to umyselne koli rychlosti, aj ked uznavam ze pri prvkoch asi necitit zavan vetra vo vlasoch, ale trust me bro je to boost :D 

+ som tym chcel vyjadrit ze v tomto pripade nam nezalezi na poradi ale iba co je v strukture, preto som zvolil set()

ale mozem to prerobit na tuple bez problemov len som chcel vysvetlit ako som na to dosiel